cosinnus_message/rocket_chat.py

```diff
@@ -66,8 +86,18 @@ def get_cached_rocket_connection(rocket_username, password, server_url, reset=Fa
             rocket_connection = None
 
     if rocket_connection is None:
-        rocket_connection = RocketChat(user=rocket_username, password=password, server_url=server_url, timeout=timeout)
-        cache.set(cache_key, rocket_connection, settings.COSINNUS_CHAT_CONNECTION_CACHE_TIMEOUT)
+        try:
+            session = requests.Session()
+            # Note: passing session to rocketchat connection makes the connection cachable.
+            rocket_connection = RocketChat(user=rocket_username, password=password, server_url=server_url,
+                timeout=timeout, session=session)
+            cache.set(cache_key, rocket_connection, settings.COSINNUS_CHAT_CONNECTION_CACHE_TIMEOUT)
+        except ConnectionError as e:
+            # When a connection error occurred disable rocketchat connections for 5 minutes to avoid overloading our
+            # webserver with pending requests.
+            set_rocket_down()
```

Comment:
Should the listener for a request timeout and the resulting `set_rocket_down` maybe also be called above this code block when `alive = rocket_connection.me().status_code == 200` is called? Because if a cached connection is retrieved, and RC goes down in that second, the call to `me()`, which is an API request, would then block, but not trigger an activation of `set_rocket_down`.

Reply:
I tried this out, as it sounded like a good idea to avoid waiting for a double timeout before detecting a downtime.

The problem is that if we restart the rocket chat service  the cached connections stop working, but we don't wont to consider the service down.